draft-ietf-taps-interface.md

```diff
@@ -1015,33 +1032,24 @@ set of send parameters not consistent with the Connection's transport
 parameters. The SendError contains an implementation-specific reference to the
 Message to which it applies.
 
-## Send Parameters {#send-params}
+## Message Send Parameters {#send-params}
 
-The Send Action takes per-Message send parameters which control how the
-contents will be sent down to the underlying Protocol Stack and transmitted.
+Each Message can be marked with Send Parameters to control how Data
+associated with the Message will be sent down to the underlying Protocol Stack 
+and transmitted. Note that these properties are per-Message, not per-Send.
+All data portions associated with a single Message share properties. For example,
+it would not make sense to have the beginning of a Message expire, but allow
+the end of a Message to still be sent.
 
-If Send Parameters should be overridden for a specific Message, an
-empty sent parameter Object can be acquired and all desired Send Parameters
-can be added to that Object. A sendParameters Object can be reused for
-sending multiple contents with the same properties.
+Parameters may be added to a Message object before the Message is used
+for sendings.
```

Comment:
Maybe comment that message parameters cannot change after Send is invoked?

Reply:
I'm wondering how this works with, e.g., "Final" messages. If Message Properties cannot be changed after Send, the application cannot reuse a Message object with an additional "Final" parameter added, but needs to create a new Message object just for this purpose, adding all properties of the previous messages to it, and "Final". Is this intended?

Reply:
If you need to do a write close with Final, you should mark that on your Message in the first send. If you need to send a partial Message that isn't necessarily the Final one, then don't mark it Final. You can later send a complete message that is final with no data.

Reply:
@chris-wood Added text